docs/package_guide.md

```diff
@@ -0,0 +1,141 @@
+# Creating packages
+
+
+## The goal
+
+The goal is to build and distribute the given program in a `.ipkg` file. The
+[toltec](https://github.com/toltec-dev/toltec) git repo's job is to pull in other repositories sourcers, invoke the compilation (if any) and then create a proper `.ipk` file out of it which is then made available [here](https://toltec.delab.re/) to opkg on your reMarkable.
+
+
+## Creating a package
+
+To create a package please follow this guide.
+
+Clone the [toltec](https://github.com/toltec-dev/toltec) git repo and make sure you're on the `testing` branch which is the default one.
+
+
+### Creating a directory for your package
+
+Go into the directory `package` and create a directory for your package. This will not determine the final name of that package but should be the same nontheless.
+
+Your package name should be **all lower case** only contain **hyphens** (`-`) at most if really necessary.
+
+
+### Adding the metadata
+
+Create a file `package`. You can either look at the other examples or create one from this template without the comments (not the vim comment):
+
+#### Your package version (`pkgver`)
+
+The package version should be using [semantic versioning](https://semver.org/). If not possible or used by your software you als may use other formats, but please do not use underscores here. An outlier that doesn't use semantic versioning is [koreader](https://github.com/toltec-dev/toltec/blob/testing/package/koreader/package).  
```

Comment:
This is not consistent with https://github.com/toltec-dev/toltec/blob/testing/docs/package.md#pkgver-required, which does not mandate semantic versioning, but rather to follow the upstream versioning scheme as closely as possible. I think the latter is preferable since we don’t want to create our own versioning scheme for each package. Also, it is important that the versions are strictly increasing (following the Debian version ordering rules), so that Opkg actually detects new versions (it will fail if a lower version than the currently installed one is published to the repo).

Reply:
So this should probably be updated to say that if no versioning system is in use, we recommend that the software adopts semantic versioning. Otherwise using a mechanism where the number increases is required.